lib/transactions/current_user_status_transaction.dart

```diff
@@ -0,0 +1,74 @@
+typedef NativeCallback = Future<bool> Function(
+    List<String> enabledPurposes, List<String> disabledPurposes, List<String> enabledVendors, List<String> disabledVendors);
+
+/// Class used to contain all the operations related to transactions.
+/// All the changes regarding the status of purposes and vendors will
+/// only be communicated to the native SDKs once the `commit` method is called.
+class CurrentUserStatusTransaction {
+  final Map<String, bool> purposesStatus = {};
```

Comment:
If possible, we should prevent clients from modifying this directly

Reply:
I've added `_` as a prefix which makes fields private.